src/app/features/reminders/reminder-card/reminder-card.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import Status, { StatusDesc } from '../../../core/enums/status.enum';
+import { Reminder } from '../../../core/models/reminder.model';
+
+@Component({
+  selector: 'app-reminder-card',
+  templateUrl: './reminder-card.component.html',
+  styleUrls: ['./reminder-card.component.scss']
+})
+export class ReminderCardComponent implements OnInit {
+  @Input() reminder!: Reminder;
+  @Input() save!: () => void;
+  @Input() reminders: Reminder[];
+  @Output() remindersUpdate = new EventEmitter<Reminder[]>();
+
+  constructor() {
+    this.reminders = [];
+  }
+
+  ngOnInit(): void {
+    this.reminders = this.getReminders();
+  }
+
+  remove(id: number) {
+    let reminder = this.reminders?.find(r => r.id == id)!;
+    if (reminder) {
+      reminder.deleted = true;
+      this.update();
+      this.save();
+    }
+  }
+
+  restore(id: number) {
+    let reminder = this.reminders?.find(r => r.id == id)!;
+    if (reminder) {
+      reminder.deleted = false;
+      this.update();
+      this.save();
+    }
+  }
+
+  delete(id: number) {
```

Comment:
Methods like restore, delete, update, etc... are actions that should be fired in this component but executed in the parent one.

Reply:
> Methods like restore, delete, update, etc... are actions that should be fired in this component but executed in the parent one.

Yes, in the begining I had them all in the parent component, and the child just triggered, I opted to move them to the child component because I wanted to leave that responsability concentrated there since everything was happening there already,.

I understand it would be better in the parent, but when I moved them all to child I was thinking that child was already being responsible for this manipulations and the parent was just providing the data and displaying it.